session/session.go

```diff
@@ -0,0 +1,34 @@
+package session
+
+import (
+	"fmt"
+	"time"
+
+	"launchpad.net/gozk"
+)
+
+var ErrNotConnected = fmt.Errorf("Unable to connect to ZooKeeper.")
+
+type ZkSession struct {
+	Connection *zookeeper.Conn
+	Events     <-chan zookeeper.Event
+}
+
+func NewZkSession(servers string) (*ZkSession, error) {
+	zkconn, zkevents, err := zookeeper.Dial(servers, 3*time.Second)
+
+	if err != nil {
+		return nil, err
+	}
+
+	select {
+	case event := <-zkevents:
+		if event.State != zookeeper.STATE_CONNECTED {
```

Comment:
Thanks for all the comments in regards to the connection management. After some more investigation I see what you mean. I give some love to the connection management and will update the PR for review.